consul-upstream-load-balancing/step4.md

```diff
@@ -0,0 +1,47 @@
+
+The load balancing policy for the datacenter applies also to the service resolution performed by ingress gateways. So, once you configured the policies for the services and tested it internally using the client service, you can introduce an ingress gateway in your configuration and the same policies will be now respected by external requests being served by your Consul datacenter.
+
+<!--Arch diagram-->
+![Consul service mesh load balancing with ingress gateway](./assets/consul-lb-envoy-ingress-gw.png)
+
+In this lab you can access the service from your browser using the [ingress gateway](https://[[HOST_SUBDOMAIN]]-8080-[[KATACODA_HOST]].environments.katacoda.com/).
+
+By refreshing the page in your web browser you can verify that the request is being balanced between both instances of the service.
+
+To test the load balancing policy on ingress gateways you can re-enable the sticky session for service resolution.
+
+You can reuse the `hash-resolver.hcl`{{open}} file and apply the policy using the `consul config` command.
+
+`docker exec server consul config write /etc/consul.d/hash-resolver.hcl`{{execute}}
+
+Example output:
+
+```
+Config entry written: service-resolver/backend
+```
+
+### Verify the policy is applied
+
```

Comment:
Do you want to add a statement about refreshing the browser to test too?

Reply:
Browser cannot pass the header so easily unless we ask them to use smt like poastman